### PR TITLE
Fix Schedule Limits Error Message

### LINKED ISSUE #9277: ScheduleTypeLimits Causing Fatal w/o Messages When Lower/Upper Values Reversed
Issue overview
--------------
Take any IDF, such as 1ZoneUncontrolled and reverse the values of the `On/Off` `ScheduleTypeLimits` object. EnergyPlus will fail without any meaningful error messages.

```
  ScheduleTypeLimits,
    On/Off,                  !- Name
    1,                       !- Lower Limit Value   <---- Reversed
    0,                       !- Upper Limit Value   <---- Reversed
    DISCRETE;                !- Numeric Type
```

The error message that comes out is below.

```
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
```

(There are no previous warnings)

SHA: 923772d985c14e803c7ae0e5348a2d595fe5739d

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
 - Fixes #9277 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
The reason for the failure is due to a mismatch in the number of arguments (`{}`) in the format string. The format string expected 6 values, but 7 were passed, so `format` just failed and throws a fatal error which is caught here: 

https://github.com/NREL/EnergyPlus/blob/923772d985c14e803c7ae0e5348a2d595fe5739d/src/EnergyPlus/api/EnergyPlusPgm.cc#L402

This seems to bring up a larger issue - we have no way of catching when we have not correctly aligned the `format` arguments and the values passed. It seems that E+ will just catch the error as shown, but not fail as it should. We probably need a custom check script to expose these issues.

**Comment by mitchute** [comment]
Ready for review.

I've got a draft script for doing some validation on these format strings, and I've found a few other issues. The script isn't ready for prime time just yet but I'll have it polished up and the issues fixed before the release date.

File: src/EnergyPlus/ScheduleManager.cc [hunk 1]
```diff
@@ -639,7 +639,7 @@ namespace ScheduleManager {
                         ShowContinueError(state, "  Other warning/severes about schedule values may appear.");
                     } else {
                         ShowSevereError(state,
-                                        format("{}=\"{}\", {} [{:.0R}] > {} [{:.0R}].",
+                                        format("{}{}=\"{}\", {} [{:.0R}] > {} [{:.0R}].",
```
Comment (Myoldmopar) [inline src/EnergyPlus/ScheduleManager.cc:642]: Clearly the correct fix to get the right number of placeholders as arguments.

**Review by Myoldmopar — APPROVED** [review]
Happy to take this.  I know your script will be a big help with the robustness of the format strings, but it doesn't have to be attached to this PR.

**Comment by Myoldmopar** [comment]
Thanks @mitchute , merging this one.

**Pull request merged**